app/Woocommerce.php

```diff
@@ -180,4 +182,48 @@ public static function get_currencies() {
 			$base_currency => ''
 		) );
 	}
+
+	/**
+	 * Update variable product price based on variations
+	 * This function is used to update the price of a variable product based on the variations
+	 * Since the price of a variable product is not stored in the product itself, we need to get the price from the variations
+	 * Task : https://app.clickup.com/t/86eprwe91
+	 * @since   1.5.0
+	 * @param   array $product_data
+	 * @param   int $product_id
+	 * @param   \WC_Product $product
+	 * @return  array
+	 */
+	public function update_variable_product_price( $product_data, $product_id, $product ) {
+
+		if ( $product->get_type() == 'variable' ) {
+
+			$first_currency = array_key_first( $product_data['price'] );
+
+			$min_price = 0;
+
+
+			// get variations
+			$variations = $product->get_children();
+
+			foreach ( $variations as $variation_id ) {
+				$variation = wc_get_product( $variation_id );
+
+				$variation_data = Product::get_instance()->generate_typesense_data( $variation );
+
+				$variation_data = apply_filters( 'blaze_wooless_get_variation_prices', $variation_data, $variation_id, $variation );
+
+				if ( $variation_data['price'][ $first_currency ] > $min_price ) {
+					$min_price = $variation_data['price'][ $first_currency ];
+
+					$product_data['price'] = $variation_data['price'];
+					$product_data['regularPrice'] = $variation_data['regularPrice'];
+					$product_data['salePrice'] = $variation_data['salePrice'];
+				}
+
+			}
```

Comment:
I would recommend we update the logic to save server usage and do the follow. 
1. use get_variation_price , get_variation_sale_price & get_variation_regular_price from https://woocommerce.github.io/code-reference/classes/WC-Product-Variable.html#method_get_variation_regular_price
2. Create a helper method where it allows to convert all prices to different currencies. Or maybe there is already and existing helper method in the aelia extension we made to get all the currency prices. 
3. use the data in step 2 to update variation prices